packages/react-native/React/DevSupport/RCTInspectorUtils.mm

```diff
@@ -18,15 +18,24 @@ @implementation RCTInspectorUtils
 
 + (CommonHostMetadata *)getHostMetadata
 {
+#if TARGET_OS_IPHONE
   UIDevice *device = [UIDevice currentDevice];
+  NSString *deviceName = [device name];
+#else
+  // macOS does not support UIDevice. Use System Configuration. This API
+  // returns a nullable value, but is non-blocking (compared with
+  // `[NSHost currentHost]`) and is ideal since deviceName is optional.
+  NSString *deviceName = [(id)SCDynamicStoreCopyComputerName(NULL, NULL) autorelease];
```

Comment:
Unless this code is running in non ARC, I would not call autorelease. However, CFStringRef can be toll-free bridged to NSString, so:
```suggestion
  NSString *deviceName = (__bridge NSString *)SCDynamicStoreCopyComputerName(nil, nil);
```